### PR TITLE
feat: Add vision instrumentation

### LINKED ISSUE #582: [vision] [python] llama-index image messages parsing

### PR DESCRIPTION
**Pull request by fjcasti1**
Closes #582

### CODE REVIEW

File: python/instrumentation/openinference-instrumentation-llama-index/examples/multimodal_stream_chat.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+endpoint = "http://127.0.0.1:6006/v1/traces"
+tracer_provider = trace_sdk.TracerProvider()
+tracer_provider.add_span_processor(SimpleSpanProcessor(OTLPSpanExporter(endpoint)))
+tracer_provider.add_span_processor(
+    span_processor=SimpleSpanProcessor(span_exporter=ConsoleSpanExporter())
+)
+
+LlamaIndexInstrumentor().instrument(tracer_provider=tracer_provider)
+OpenAIInstrumentor().instrument(tracer_provider=tracer_provider)
```
Comment (RogerHYang) [inline python/instrumentation/openinference-instrumentation-llama-index/examples/multimodal_stream_chat.py:20]: let's keep the examples focused solely on llama index
```suggestion
```

File: python/instrumentation/openinference-instrumentation-llama-index/tests/openinference/instrumentation/llama_index/test_handler_multimodal.py [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+            resp_gen = llm.stream_chat([message])
+            _ = [r for r in resp_gen]
+        else:
+            _ = llm.chat([message])
+
+    async def achat(message: ChatMessage) -> None:
+        if is_stream:
+            resp_gen = await llm.astream_chat([message])
+            x = [r async for r in resp_gen]
+            print(x)
```
Comment (RogerHYang) [inline python/instrumentation/openinference-instrumentation-llama-index/tests/openinference/instrumentation/llama_index/test_handler_multimodal.py:151]:
```suggestion
```

**Review by RogerHYang — APPROVED** [review]
well done!

File: python/instrumentation/openinference-instrumentation-llama-index/src/openinference/instrumentation/llama_index/_handler.py [hunk 3]
```diff
@@ -910,6 +915,37 @@ def _asdict(obj: Any) -> Any:
             return repr(obj)
 
 
+def _get_attributes_from_message_content(
+    content: Mapping[str, Any],
+) -> Iterator[Tuple[str, AttributeValue]]:
+    content = dict(content)
+    type_ = content.pop("type")
```
Comment (RogerHYang) [inline python/instrumentation/openinference-instrumentation-llama-index/src/openinference/instrumentation/llama_index/_handler.py:922]: i'm surprised the type checker didn't pick this up, but `Mapping` is not supposed to be mutable.
```suggestion
    type_ = content.get("type")
```

Comment (fjcasti1) [reply python/instrumentation/openinference-instrumentation-llama-index/src/openinference/instrumentation/llama_index/_handler.py:922]: Oh good call

File: python/instrumentation/openinference-instrumentation-llama-index/src/openinference/instrumentation/llama_index/_handler.py [hunk 4]
```diff
@@ -910,6 +915,37 @@ def _asdict(obj: Any) -> Any:
...
+        if image := content.pop("image_url"):
+            for key, value in _get_attributes_from_image(image):
+                yield f"{MessageContentAttributes.MESSAGE_CONTENT_IMAGE}.{key}", value
+
+
+def _get_attributes_from_image(
+    image: Mapping[str, Any],
+) -> Iterator[Tuple[str, AttributeValue]]:
+    image = dict(image)
+    if url := image.pop("url"):
```
Comment (RogerHYang) [inline python/instrumentation/openinference-instrumentation-llama-index/src/openinference/instrumentation/llama_index/_handler.py:938]:
```suggestion
    if url := image.get("url"):
```

File: python/instrumentation/openinference-instrumentation-llama-index/examples/multimodal_stream_chat.py (hunk 1 above)
Comment (fjcasti1) [reply python/instrumentation/openinference-instrumentation-llama-index/examples/multimodal_stream_chat.py]: Done

**Pull request merged**